graylog2-server/src/main/java/org/graylog2/indexer/searches/Searches.java

```diff
@@ -805,16 +813,27 @@ private QueryBuilder standardAggregationFilters(TimeRange range, String filter)
         return Optional.empty();
     }
 
-    public Set<String> determineAffectedIndices(TimeRange range,
-                                                @Nullable String filter) {
+    @VisibleForTesting
+    Set<String> determineAffectedIndices(TimeRange range, @Nullable String filter) {
         final Set<IndexRange> indexRanges = determineAffectedIndicesWithRanges(range, filter);
-        return indexRanges.stream()
-                .map(IndexRange::indexName)
-                .collect(Collectors.toSet());
+        if(indexRanges.size() > MAX_INDICES_PER_QUERY) {
+            return indexRanges.stream()
+                    .map(IndexRange::indexName)
+                    .map(indexSetRegistry::getForIndex)
+                    .flatMap(o -> o.map(java.util.stream.Stream::of).orElseGet(java.util.stream.Stream::empty))
+                    .map(IndexSet::getWriteIndexAlias)
```

Comment:
I guess you wanted to use `IndexSet#getIndexWildcard()` here? Using `graylog_deflector` doesn't make sense here. AFAIK